packages/components/src/components/gif.tsx

```diff
@@ -179,6 +204,23 @@ const Gif = ({
         setHasFiredSeen(false)
     }, [gif.id])
 
+    const stopTracking = () => {
+        // if we have a moat ad number
+        if (moatAdNumber.current !== undefined) {
+            // stop tracking
+            moat.stopTracking(moatAdNumber.current)
+            // remove the moat ad number
+            moatAdNumber.current = undefined
+        }
+    }
+
+    // if this component goes from showing an ad to not an ad
+    useEffect(() => {
+        if (!shouldShowMedia) {
+            stopTracking()
+        }
```

Comment:
@nicksantan you want to `stopTracking` when you unmount, so returning `return () => stopTracking()` would be wise

Reply:
Thanks Gianni!